### DISCUSSION TITLE
[QUESTION] Getting the tagged user

### DISCUSSION THREAD

**Discussion by AnimusXCASH** [python-telegram-bot/python-telegram-bot#2726] [Q&A]
### Issue I am facing

Hi, 

I have been trying to find the way on how to get the tagged user from command as an object, however unsuccessfully. If I check for Message entity MENTION when executing command with structure: 
`/gift @teegram.User`

I get a NoneType when trying to access tagged users ID.

Code structure attached.
                    
Also I tried the other way with accessing arguments, also that did not work.  Tagged user is not recognized therefore returning **TypeError: 'NoneType' object is not subscriptable**

Any solution or guidance would be appreciated a lot. 

Thank you for your time and answer on my request

brg

 

### Related part of your code

```python
def gift(self, update, context):
        msg = update.effective_message
        if msg.entities and msg.parse_entities([MessageEntity.TEXT_MENTION,MessageEntity.MENTION]):
            entities = msg.parse_entities([MessageEntity.TEXT_MENTION,MessageEntity.MENTION])

            for ent in entities:
                if ent.type == MessageEntity.TEXT_MENTION:
                    user_id = ent.user.id
                    fst_name = ent.user.first_name
                    print(user_id)
                    print(fst_name)
                elif ent.type == MessageEntity.MENTION:
                    print(ent.user.id)
```

### Operating System

Windows

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 13.7
Bot API 5.3
certifi 2021.05.30
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

**Comment by Bibo-Joshi** [comment]
Hi. `MessageEntity.user` will only be present for `ME.TEXT_MENTION` entities, not for `ME.MENTION`. That's just a limitation by Telegram.

**Comment by AnimusXCASH** [comment]
So there is really no way to get the user id if he is tagged inside user post?

**Reply by Bibo-Joshi** [reply]
There's no method in the Bot API that converts a username into a chat id, no.

**Comment by harshil21** [comment]
Well there is [ptbcontrib/username_to_chat_api](https://github.com/python-telegram-bot/ptbcontrib/tree/main/ptbcontrib/username_to_chat_api) if you'd like to use that.

**Reply by AnimusXCASH** [reply]
Hmm interesting. I'l give it a shot. I was already exploring the option to use string in telethon just to get ID 

Thank you @harshil21
